lib/bindevict.go

```diff
@@ -85,41 +85,44 @@ func (entry *BindThrottle) decrAllowEveryX(y int) {
 		return
 	}
 	entry.AllowEveryX = 0
-	GetBindEvict().lock.Lock()
-	defer GetBindEvict().lock.Unlock()
+}
+func (entry *BindThrottle) incrAllowEveryX() {
+	if logger.GetLogger().V(logger.Warning) {
+		info := fmt.Sprintf("hash:%d bindName:%s val:%s prev:%d", entry.Sqlhash, entry.Name, entry.Value, entry.AllowEveryX)
+		logger.GetLogger().Log(logger.Warning, "bind throttle incr", info)
+	}
+	entry.AllowEveryX = 3*entry.AllowEveryX + 1
+	if entry.AllowEveryX > 10000 {
+		entry.AllowEveryX = 10000
+	}
+}
+
+func (bindEvict *BindEvict) updateThrottle(entry *BindThrottle) {
```

Comment:
This function needs to be used with caution (hold the lock when updating the copy), the new struct has a different lock.